packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel/utils/exception.dart

```diff
@@ -10,13 +10,13 @@ import '../../firebase_auth_exception.dart';
 
 /// Catches a [PlatformException] and converts it into a [FirebaseAuthException]
 /// if it was intentionally caught on the native platform.
-Exception convertPlatformException(Object exception) {
+Object convertPlatformException(Object exception) {
   if (exception is! Exception || exception is! PlatformException) {
+    // TODO(rrousselGit): Is this dead code?
     return exception;
```

Comment:
This code cannot be assigned to `Exception`, so I changed the return type to `Object`.

But that raise a question: Why do we even need this code? It probably was never reached before, as we'd otherwise get a "XX cannot be assigned to Exception" thrown

Reply:
I can't think of any reason why this was needed